pandas/tests/indexes/test_numeric.py

```diff
@@ -259,6 +259,12 @@ def test_astype(self):
         for dtype in ['M8[ns]', 'm8[ns]']:
             self.assertRaises(TypeError, lambda: i.astype(dtype))
 
+        # GH 13149
+        # a NaN astype int
```

Comment:
Do you mean this conversion:
```
pd.DatetimeIndex([1,2,np.nan]).astype(int)
Out[16]: array([                   1,                    2, -9223372036854775808])
```
Its behaviour is unchanged. I didn't touch other indexes.